src/operation.js

```diff
@@ -19,15 +23,18 @@ export class Operation {
     * @returns {xdr.CreateAccountOp}
     */
     static createAccount(opts) {
-        if (!opts.destination) {
-            throw new Error("Must provide a destination for a payment operation");
+        if (!Account.isValidAddress(opts.destination)) {
+            throw new Error("destination is invalid");
         }
-        if (!opts.startingBalance) {
-            throw new Error("Must provide a starting balance");
+        if (!isString(opts.startingBalance)) {
+            throw new TypeError('startingBalance argument must be of type String');
+        }
+        if (isEmpty(opts.startingBalance)) {
+            throw new TypeError('startingBalance argument must not be empty');
         }
         let attributes = {};
         attributes.destination  = Keypair.fromAddress(opts.destination).accountId();
-        attributes.startingBalance = Hyper.fromString(String(opts.startingBalance));
+        attributes.startingBalance = Hyper.fromString(opts.startingBalance);
```

Comment:
Note:  We still haven't updated js-stellar-base to interpret amounts within our conventions for client libraries and horizon (i.e.  7 fixed decimal places)